nox/virtualenv.py

```diff
@@ -134,59 +140,46 @@ def _resolved_interpreter(self):
             self._resolved = sys.executable
             return self._resolved
 
+        # Otherwise we need to divine the path to the interpreter. This is
+        # designed to accept strings in the form of "2", "2.7", "2.7.13",
+        # "2.7.13-32", "python2", "python2.4", etc.
+        xy_version = ""
+        cleaned_interpreter = self.interpreter
+
         # If this is just a X, X.Y, or X.Y.Z string, extract just the X / X.Y
         # part and add Python to the front of it.
         match = re.match(r"^(?P<xy_ver>\d(\.\d)?)(\.\d+)?$", self.interpreter)
-        xy_version = ""
-        cleaned_interpreter = self.interpreter
         if match:
             xy_version = match.group("xy_ver")
             cleaned_interpreter = "python{}".format(xy_version)
 
-        # The value of ``cleaned_interpreter`` may be a valid script on the ``PATH``
-        # or may be a full path to an interpreter in the filesystem; accept this.
+        # If the cleaned interpreter is on the PATH, go ahead and return it.
         if py.path.local.sysfind(cleaned_interpreter):
             self._resolved = cleaned_interpreter
             return self._resolved
 
+        # The rest of this is only applicable to Windows, so if we don't have
+        # an interpreter by now, raise.
+        if _SYSTEM != "Windows":
+            self._resolved = InterpreterNotFound(self.interpreter)
+            raise self._resolved
+
         # Allow versions of the form ``X.Y-32`` for Windows.
-        if not xy_version:
-            match = re.match(r"^\d\.\d-32?$", cleaned_interpreter)
-            if match:
-                xy_version = cleaned_interpreter
+        match = re.match(r"^\d\.\d-32?$", cleaned_interpreter)
```

Comment:
@theacodes It's possible that the user has already specified `python2.7` in which case calling `re.match` here is "wasted computation".

Reply:
It's fine.